### PR TITLE
fix(slider): snap to mark

### PR DESCRIPTION
**Pull request by merlingson**
### Describe your changes

- When a mark is the target of a click event, the slider will now snap to the value of the closest mark and not the specific clientX value.
-  Add example to slider page where both steps and marks are combined

### Issue ticket number and link

- N/A

### Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have verified that the code builds perfectly fine on my local system
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have verified that my code follows the style already available in the repository
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by NiklasRamstrom — APPROVED** [review]
LGTM

**Comment by NiklasRamstrom** [comment]
Looks good.
Before merging, you need to verify your commit, as this is a requirement in this repository. You can read more here:
https://docs.github.com/en/authentication/managing-commit-signature-verification

**Comment by merlingson** [comment]
> Looks good. Before merging, you need to verify your commit, as this is a requirement in this repository. You can read more here: https://docs.github.com/en/authentication/managing-commit-signature-verification

Thank you! The commit should be verified now. The workflows requires additional approval

**Pull request merged**
